### PR TITLE
Update Docker_Swarm workflow to deploy swarm nodes on both Vms of a mixed node source (trydev_trydev2) + Update worklfow chapeau Hbase + Update description in Yarn_Pi_example

### PR DESCRIPTION
**Pull request by rsaidane**

### CODE REVIEW

**Review by fviale — APPROVED** [review]
LGTM

File: DockerSwarm/resources/catalog/Docker_Swarm.xml [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+    <variable name="TARGETED_NB_PA_NODES" value="3" model="PA:NOT_EMPTY_STRING" description="The number of targeted deployment nodes."  advanced="false" hidden="false"/>
+    <variable name="TARGETED_NB_PA_HOSTS" value="1" model="PA:Integer" description="Minimum number of machines (hosts) that will be contained in the docker swarm infrastructure. This should correspond to the number of machines (VMs) contained in the target node source."  advanced="true" hidden="false"/>
+    <variable name="HOST_CHECK" value="" model="PA:SPEL(T(Integer).parseInt(variables[&#x27;TARGETED_NB_PA_NODES&#x27;]) &gt;= T(Integer).parseInt(variables[&#x27;TARGETED_NB_PA_HOSTS&#x27;]))" description="Check NB_NODES &gt;= NB_HOSTS"  advanced="false" hidden="true"/>
   </variables>
   <description>
     <![CDATA[ Create a Docker Swarm. TARGETED_PA_NODE_SOURCE_NAME refers to the targeted node source from which TARGETED_NB_PA_NODES ProActive nodes will be reserved for the Docker Swarm. ]]>
```
Comment (mbenguig) [inline DockerSwarm/resources/catalog/Docker_Swarm.xml:13]: You need to update the wkw description.

File: Hbase/resources/catalog/Docker_Swarm_HDFS_Zookeeper_Hbase_Service.xml [hunk 2]
```diff
@@ -3,12 +3,14 @@
...
+        <variable name="TARGETED_NB_HOSTS" value="1" model="PA:Integer" description="Minimum number of machines (hosts) that will be contained in the docker swarm infrastructure. This should correspond to the number of machines (VMs) contained in the target node source."  advanced="true" hidden="false"/>
+        <variable name="HOST_CHECK" value="" model="PA:SPEL(T(Integer).parseInt(variables[&#x27;NB_HBASE_HDFS_WORKERS&#x27;]) &gt;= T(Integer).parseInt(variables[&#x27;TARGETED_NB_HOSTS&#x27;]))" description="Check NB_NODES &gt;= NB_HOSTS"  advanced="false" hidden="true"/>
     </variables>
     <description>
-        <![CDATA[ Deploy a Docker Swarm-HDFS-Zookeeper-Hbase platform of nb_hbase_hdfs_workers region servers and hdfs data nodes. node_source_name refers to the targeted node source of the Hbase-HDFS deployment. ]]>
+        <![CDATA[ Deploy a Docker Swarm-HDFS-Zookeeper-Hbase platform of NB_HBASE_HDFS_WORKERS region servers and hdfs data nodes. node_source_name refers to the targeted node source of the Hbase-HDFS deployment. ]]>
```
Comment (mbenguig) [inline Hbase/resources/catalog/Docker_Swarm_HDFS_Zookeeper_Hbase_Service.xml:13]: You need to update the description (new wkw params + behavior)

File: DockerSwarm/resources/catalog/Docker_Swarm.xml [hunk 3]
```diff
@@ -132,31 +137,36 @@ println ""
...
             <![CDATA[
-// Select the nodes of the targeted node source except the swarm manager node, since this latter is already tokenized
-selected = System.getProperty("proactive.node.nodesource").equals(variables.get("TARGETED_PA_NODE_SOURCE_NAME"))
+def task_iteration_id = variables.get("PA_TASK_ITERATION") as Integer
+def target_nb_nodes = variables.get("TARGETED_NB_PA_NODES") as Integer
+def target_nb_hosts = variables.get("TARGETED_NB_PA_HOSTS") as Integer
+
+chosenNodeSource = System.getProperty("proactive.node.nodesource").equals(variables.get("TARGETED_PA_NODE_SOURCE_NAME"))
+
+selected = chosenNodeSource && (variables.get("CURRENT_HOSTS").size() == target_nb_hosts || (variables.get("CURRENT_HOSTS").size() < target_nb_hosts  && !variables.get("CURRENT_HOSTS").contains(nodehost) ))
```
Comment (mbenguig) [inline DockerSwarm/resources/catalog/Docker_Swarm.xml:169]: Could you add a comment ? Hard to understand this new node selection policy.

File: YarnOrchestration/resources/catalog/Yarn_Pi_Example.xml [hunk 4]
```diff
@@ -3,9 +3,9 @@
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xmlns="urn:proactive:jobdescriptor:3.13" xsi:schemaLocation="urn:proactive:jobdescriptor:3.13 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.13/schedulerjob.xsd"  name="YARN_Pi" projectName="01. Basic MapReduce Examples" priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2"  >
     <variables>
-        <variable name="nb_maps" value="16" model=""/>
-        <variable name="nb_samples" value="10000" model=""/>
-        <variable name="yarn_service_instance_id" value="xx"  model="PA:NOT_EMPTY_STRING"/>
+        <variable name="nb_maps" value="16" model="" description="Is the number of maps to create," group="" advanced="false" hidden="false"/>
```
Comment (mbenguig) [inline YarnOrchestration/resources/catalog/Yarn_Pi_Example.xml:6]: -> "The number of maps to create"

File: DockerSwarm/resources/catalog/Docker_Swarm.xml (hunk 3 above)
Comment (fviale) [reply DockerSwarm/resources/catalog/Docker_Swarm.xml:173]: `// CURRENT_HOSTS contains the set of hosts (Vms) already registered in this deployment.
// target_nb_hosts is the number of hosts we want to deploy on
// if target_nb_hosts is not yet reached, we try to find a host that is not already registered
`

also in the test, change:
`variables.get("CURRENT_HOSTS").size() == target_nb_hosts` to `variables.get("CURRENT_HOSTS").size() >= target_nb_hosts`

**Pull request merged**